File: python_ruby/multiple_inheritance/2.rb
module CalcMultiply
  def multiply()
    return @value1 * @value2
  end
end

module CalcDivide
  def divide()
    return @value1 / @value2
  end
end

class Calc
  include CalcMultiply, CalcDivide

  def initialize(value1, value2)
    if value1.is_a?(Integer)
      @value1 = value1
    end
    if value2.is_a?(Integer)
      @value2 = value2
    end
  end

  def add()
    return @value1 + @value2
  end

  def setValue(value)
    if value.is_a?(Integer)
      @value1 = value
    end
  end
  def getValue1()
    return @value1
  end
end

c = Calc.new(100,10)
p c.getValue1()
p c.multiply()
p c.divide()
p c.add()

File: python_ruby/multiple_inheritance/2.py
class CalcMultiply():
    def multiply(self):
        return self.value1 * self.value2

class CalcDivide():
    def divide(self):
        return self.value1 / self.value2

class Calc(CalcMultiply, CalcDivide):
    def __init__(self, value1, value2):
        if isinstance(value1,int):
            self.value1 = value1
        if isinstance(value2,int):
            self.value2 = value2

    def add(self):
        return self.value1 + self.value2

    def subtract(self):
        return self.value1 - self.value2

    def setValue1(self, value1):
        if isinstance(value1,int):
            self.value1 = value1

    def getValue(self):
        return self.value1

c = Calc(100,10)
print(c.add())
print(c.multiply())
print(c.subtract())
print(c.divide())
